src/infra/src/table/migration/m20241217_155000_populate_alerts_table.rs

```diff
@@ -0,0 +1,1035 @@
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Populates the `alerts` table by transforming unstructured alert records from
+//! the `meta` table.
+
+use itertools::{self, Itertools};
+use sea_orm::{
+    ActiveValue::NotSet, EntityTrait, FromQueryResult, Paginator, PaginatorTrait, SelectModel, Set,
+    Statement, TransactionTrait,
+};
+use sea_orm_migration::prelude::*;
+use svix_ksuid::KsuidLike;
+
+/// The value of the flag used to indicate the alerts folder type in the folders
+/// table.
+const ALERTS_FOLDER_TYPE: i16 = 1;
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        let txn = manager.get_connection().begin().await?;
+
+        // Select all orgs that have an alert. Create a default alert folder for
+        // each org. Migrate pages of 100 records at a time to avoid loading too
+        // many records into memory.
+        let mut org_pages = OrgWithAlert::paginate(&txn, 100);
+        while let Some(orgs) = org_pages.fetch_and_next().await? {
+            let folders = orgs
+                .into_iter()
+                .map(|o| folders_table::ActiveModel {
+                    id: NotSet,
+                    org: Set(o.org_name().to_owned()),
+                    folder_id: Set("default".to_owned()),
+                    name: Set("default".to_owned()),
+                    description: Set(Some("default".to_owned())),
+                    r#type: Set(ALERTS_FOLDER_TYPE),
+                })
+                .collect_vec();
+            folders_table::Entity::insert_many(folders)
+                .exec(&txn)
+                .await?;
+        }
+
+        // Select each alert from the meta table joined on its associated
+        // default folder. For each alert from the meta table insert a
+        // corresponding record in the new alerts table. Migrate pages of 100
+        // records at a time to avoid loading too many records into memory.
+        let mut meta_pages = MetaAlertWithFolder::paginate(&txn, 100);
+        while let Some(metas) = meta_pages.fetch_and_next().await? {
+            let alerts_rslt: Result<Vec<_>, DbErr> = metas
+                .into_iter()
+                .map(|m| {
+                    let m: alerts_table::ActiveModel = m.try_into().map_err(DbErr::Migration)?;
+                    Ok(m)
+                })
+                .collect();
+            let alerts = alerts_rslt?;
+            alerts_table::Entity::insert_many(alerts).exec(&txn).await?;
```

Comment:
This isn't a bad suggestion, but for a migration like this we do want the whole operation to fail if even one conversion fails. Additionally, I don't think that logging every failure is particularly important in this case, though this would be a good strategy if it were.